### PR TITLE
#308 put CaM sources into zip archive

### PR DESCRIPTION
**Pull request by Raleksan**
Work under #308: Adding CaM sources to a zip archive.

Some notes:

- The path to clone the repo is `${TARGET}/github/self/${name}` to use a common command for all repos, removing the `.git` directories. 
- A user may be using an out-of-date version of CaM, in which case they will keep an archive with a CaM repo of a newer version.

### CODE REVIEW

**Comment by yegor256** [comment]
@Raleksan what about a unit test for this new feature? Simply put, how do you know that it works?

**Comment by Raleksan** [comment]
I think adding a test for this feature is achievable for me.

As I understand, the tests for the scripts in the `/steps` directory are located in `/test/steps` and to add a new test I need to edit `test/steps/test-zip.sh`.

Am I getting this right?

p.s. Do I always need to ping you with @***? Or did you get the notification anyway?

**Comment by yegor256** [comment]
@Raleksan you are right, the tests are in the `test/steps/test-zip.sh`. Also, read this please: https://www.yegor256.com/2024/04/01/ping-me-please.html

**Comment by Raleksan** [comment]
@yegor256 I added a new test, you can review it.

File: steps/zip.sh [hunk 1]
```diff
@@ -36,6 +36,12 @@ if [ -e "${zip}" ]; then
     exit
 fi
 
+cam_repo_target_dir="${TARGET}/github/self/${name}"
+
+if [ ! -d "${cam_repo_target_dir}" ]; then
+    git clone --depth 1 https://github.com/yegor256/cam.git "${cam_repo_target_dir}"
```
Comment (yegor256) [inline steps/zip.sh:42]: @Raleksan I believe, it is reasonable to delete the `"${cam_repo_target_dir}/.git"` directory after the clone

Comment (Raleksan) [reply steps/zip.sh:42]: @yegor256 I know that I need to delete `.git`, therefore I locate it at `${TARGET}/github/self/${name}` to delete `.git` directory with all others git directories, using already existing command, or it is better to do it separately and explicitly?

Comment (yegor256) [reply steps/zip.sh:42]: @Raleksan this location is wrong, I believe. Let's put it into `${TARGET}/cam-sources` instead

Comment (Raleksan) [reply steps/zip.sh:42]: @yegor256 I changed the location as you remarked and improved the test.

Can you please review it again?

**Comment by yegor256** [comment]
@Raleksan not all tests pass after your changes. Try to run `make test` locally.

**Comment by Raleksan** [comment]
@yegor256

I have analyzed the pipeline run errors and found that both failures depend on the same test [`test-integration.sh`](https://github.com/yegor256/cam/blob/master/tests/after/test-integration.sh).

I checked out [`test-integration.sh`](https://github.com/yegor256/cam/blob/master/tests/after/test-integration.sh) and here is my interpretation, could you please clarify?

The actions of `test-integration.sh`:
- Initially it uses https://github.com/yegor256/tojos as a single repo to collect metrics (lines 29-44).
- After analysis, it checks that all artifacts (hashes, reports, etc) exist (lines 45-48)
- Checks the number of entities contained in the `TARGET` directory **[this step raises an error]** (line 49)
- Checks if certain files exist (lines 50-66)

I'm interested in line 49:
> `test ‘$(find “${TARGET}” -maxdepth 1 | wc -l | xargs)’ = 10`.
 
It's unclear to me from where we inherit this ‘magic number’.

For this PR, I added 1 entity to `TARGET` which is CaM sources, which I think is causing the error in this test.

p.s. These are just my thoughts without running tests because I faced a bug while building a Docker container locally, I'm going to investigate and try to fix this problem or open an issue if  if it's a real bug

**Comment by yegor256** [comment]
@Raleksan here, 10 is the number of highest-level elements we expect to have in the array. It's indeed a magic number. I believe, you should just change it to 11 and the test will pass.

**Comment by yegor256** [comment]
@Raleksan if you have problem with Docker, please report an issue, we will try to resolve it

**Comment by yegor256** [comment]
@Raleksan thanks!

**Pull request merged**